Tutorials/drawing-options/drawing-options-functions-using-bitmap.md

```diff
@@ -0,0 +1,176 @@
+---
+title: "Drawing Options functions using DrawBitmap"
+description: Learn how to use drawing options functions in Splashkit and combine them to create dynamic and visually stunning graphics.
+author: Sharvani Kandala
+date: 05/05/2024
+tags: programming, graphics, Splashkit, tutorials.
+---
+
+## Window Setup
+
+To begin utilizing drawing options functions in Splashkit, set up the game window as follows:
+
+1. Import the necessary Splashkit headers at the beginning of your code:
+
+   ```cpp
+   #include "splashkit.h"
+   ```
+
+2. In the `main` function, configure the game window:
+
+   ```cpp
+    int main()
+    {
+       open_window("Drawing", 800, 600);
+       return 0;
+    }
+    ```
+
+3. Function to Draw the Bitmap:
+
+   ```cpp
+   void draw_bitmap(bitmap bmp, double x, double y, drawing_options opts);
+   ```
+
+- `bmp`: The bitmap to draw.
+- `x`: The x-coordinate for the top-left corner of the bitmap.
+- `y`: The y-coordinate for the top-left corner of the bitmap.
+- `opts`: Drawing options to apply.
+
+## Drawing Options Functions
+
+## Example 1: Applying Option Part Bmp
+
+In this example, we'll demonstrate how to use 'Option Part Bmp' to draw a section of a bitmap, effectively cropping the image.
+
+1. **Function Syntax:**
+
+ ```cpp
+  drawing_options option_part_bmp(int x, int y, int width, int height);
+```
+
+- `x`: The x-coordinate for the top-left corner of the part of the bitmap to draw.
+- `y`: The y-coordinate for the top-left corner of the part of the bitmap to draw.
+- `width`: The width of the part of the bitmap to draw.
+- `height`: The height of the part of the bitmap to draw.
+
+1. **Example Usage:**
+
+   For instance, draw a window with dimensions of `800` pixels width and `600` pixels height. Here is the code snippet where we load a bitmap and applies 'Option Part Bmp' to draw only a 100x100 region of it at coordinates (100, 100).`bitmap bmp` loads a bitmap image named `my_bitmap` from the file `bitmap.png`. The bitmap is stored in the variable bmp for later use.
+
+   ```cpp
+   #include "spalshkit.h"
+   int main()
+    {
+       open_window("Drawing", 800, 600);
+
+       bitmap bmp = load_bitmap("my_bitmap", "bitmap.png");
+       drawing_options opts = option_part_bmp(0, 0, 100, 100);
+
+       draw_bitmap(bmp, 100, 100, opts);
+
+       delay(5000); // Pause for observation
+
+       close_all_windows();
+       return 0;
+    }
+
+## Example 2: Applying Option Rotate Bmp
```

Comment:
This heading should be H3 (### instead of ##) as it is in the group of the heading above the first example section.